tutorials/roostats/FourBinInstructional.py

```diff
@@ -0,0 +1,327 @@
+wspace.factory("Poisson::onbar(nonbar[0,10000], bbar[1000,500,2000])")
+wspace.factory("Poisson::offbar(noffbar[0,1000000], prod::lambdaoffbar(bbar, tau))")
+wspace.factory("Gaussian::mcCons(rhonom[1.,0,2], rho, sigma[.2])")
+wspace.factory("PROD::model(on,off,onbar,offbar,mcCons)")
+wspace.defineSet("obs", "non,noff,nonbar,noffbar,rhonom")
+
+wspace.factory("Uniform::prior_poi({s})")
+wspace.factory("Uniform::prior_nuis({b,bbar,tau, rho})")
+wspace.factory("PROD::prior(prior_poi,prior_nuis)")
+
+# ----------------------------------
+# Control some interesting variations
+# define parameers of interest
+# for 1-d plots
+wspace.defineSet("poi", "s")
+wspace.defineSet("nuis", "b,tau,rho,bbar")
+
+# for 2-d plots to inspect correlations:
+# wspace.defineSet("poi","s,rho")
+
+# test simpler cases where parameters are known.
+# wspace["tau"].setConstant()
+# wspace["rho"].setConstant()
+# wspace["b"].setConstant()
+# wspace["bbar"].setConstant()
+
+# inspect workspace
+# wspace.Print()
+
+# ----------------------------------------------------------
+# Generate toy data
+# generate toy data assuming current value of the parameters
+# import into workspace.
+# add Verbose() to see how it's being generated
+data = wspace["model"].generate(wspace.set("obs"), 1)
+# data.Print("v")
+wspace.Import(data)
+
+# ----------------------------------
+# Now the statistical tests
+# model config
+modelConfig = ROOT.RooStats.ModelConfig("FourBins")
+modelConfig.SetWorkspace(wspace)
+modelConfig.SetPdf(wspace["model"])
+modelConfig.SetPriorPdf(wspace["prior"])
+modelConfig.SetParametersOfInterest(wspace.set("poi"))
+modelConfig.SetNuisanceParameters(wspace.set("nuis"))
+wspace.Import(modelConfig)
+wspace.writeToFile("FourBin.root")
+
+# -------------------------------------------------
+# If you want to see the covariance matrix uncomment
+# wspace["model"].fitTo(data)
+
+# use ProfileLikelihood
+plc = ROOT.RooStats.ProfileLikelihoodCalculator(data, modelConfig)
+plc.SetConfidenceLevel(0.95)
+plInt = plc.GetInterval()
+msglevel = ROOT.RooMsgService.instance().globalKillBelow()
+ROOT.RooMsgService.instance().setGlobalKillBelow(ROOT.RooFit.FATAL)
+plInt.LowerLimit(wspace["s"])  # get ugly print out of the way. Fix.
+ROOT.RooMsgService.instance().setGlobalKillBelow(msglevel)
+
+# use FeldmaCousins (takes ~20 min)
+fc = ROOT.RooStats.FeldmanCousins(data, modelConfig)
+fc.SetConfidenceLevel(0.95)
+# number counting: dataset always has 1 entry with N events observed
+fc.FluctuateNumDataEntries(False)
+fc.UseAdaptiveSampling(True)
+fc.SetNBins(40)
+fcInt = ROOT.RooStats.PointSetInterval()
+if doFeldmanCousins:  # takes 7 minutes
+    fcInt = fc.GetInterval()
+
+# use BayesianCalculator (only 1-d parameter of interest, slow for this problem)
+bc = ROOT.RooStats.BayesianCalculator(data, modelConfig)
+bc.SetConfidenceLevel(0.95)
+bInt = ROOT.RooStats.SimpleInterval()
+if doBayesian and wspace.set("poi").getSize() == 1:
```

Comment:
```suggestion
if doBayesian and len(wspace.set("poi")) == 1:
```
We have Pythonizations in place that allow you to use the more pythonic `len()` everywhere we have `size()`, `getSize()`, or `GetSize()` in C++.